src/tests/end-to-end/tests/popup/adhoc-panel.test.ts

```diff
@@ -65,6 +67,34 @@ describe('Ad hoc tools', () => {
         expect(results).toMatchSnapshot();
     });
 
+    it.each(['Automated checks', 'Landmarks', 'Headings', 'Color'])(
+        'should display the pinned target page visualizations when enabling the "%s" toggle',
+        async (toggleAriaLabel: string) => {
+            await gotoAdhocPanel();
+
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
```

Comment:
Once this PR is merged, I can use this piece of code to check tabstop visualizations for my test as well. Thanks for doing this. 👍